presto-main/src/main/java/io/prestosql/operator/HttpPageBufferClient.java

```diff
@@ -622,22 +623,22 @@ public PagesResponse handle(Request request, Response response)
                     catch (RuntimeException | IOException e) {
                         // Ignored. Just return whatever message we were able to decode
                     }
-                    throw new PageTransportErrorException(format("Expected response code to be 200, but was %s:%n%s", response.getStatusCode(), body.toString()));
+                    throw new PageTransportErrorException(fromUri(uri), format("Expected response code to be 200, but was %s:%n%s", response.getStatusCode(), body.toString()));
                 }
 
                 // invalid content type can happen when an error page is returned, but is unlikely given the above 200
                 String contentType = response.getHeader(CONTENT_TYPE);
                 if (contentType == null) {
-                    throw new PageTransportErrorException(format("%s header is not set: %s", CONTENT_TYPE, response));
+                    throw new PageTransportErrorException(fromUri(uri), format("%s header is not set: %s", CONTENT_TYPE, response));
                 }
                 if (!mediaTypeMatches(contentType, PRESTO_PAGES_TYPE)) {
-                    throw new PageTransportErrorException(format("Expected %s response from server but got %s", PRESTO_PAGES_TYPE, contentType));
+                    throw new PageTransportErrorException(fromUri(uri), format("Expected %s response from server but got %s", PRESTO_PAGES_TYPE, contentType));
                 }
 
-                String taskInstanceId = getTaskInstanceId(response);
-                long token = getToken(response);
-                long nextToken = getNextToken(response);
-                boolean complete = getComplete(response);
+                String taskInstanceId = getTaskInstanceId(response, uri);
```

Comment:
do we need to pass `uri` to all those methods? We are catching `PageTransportErrorException` below anyway and add remote host information there. IMO that should be eough.

Reply:
I didn't do it that way, as that would require creating another constructor of `PageTransportErrorException` and pass remoteHost as null to its parent. I thought we would not want to do this (pass remoteHost as null for transport exceptions). I will change it if that is okay.

Reply:
Yeah... Having `remoteHost` nullable is kinda ugly. So maybe we can live with this extra parameter passed to those helper methods. @electrum ?

@sanchitkashyap  Can you also comment on motivation for the change? The only final PTEE which is sent back to user is constructed in the `catch` clause below.
And we already have whole uri as part of an exception message there. Is that not enough?

Reply:
@losipiuk  i have updated the description to explain the motivation.